cirq-core/cirq/protocols/json_test_data/GridDeviceMetadata.json

```diff
@@ -208,5 +272,18 @@
       "row": 10,
       "col": 10
     }
+  ],
+  "compilation_target_gatesets": [
```

Comment:
Don't change existing json test. Add a new test instead so that the existing json deserialization works as expected.

Reply:
We have to at least set `compilation_target_gatesets` to `[]`, otherwise the test will fail saying the JSON serialization of the instance from the repr file does not match the JSON file. So IIUC we can't deserialization backward compatibility anyway. I could perhaps add an _inward file instead and keep the old entry there.

Reply:
Ah I see, my bad. Let's add an `_inward` file instead.

Reply:
Added a `json_inward` which is a copy of the old `json`, and the corresponding `repr_inward`.